src/Compilers/CSharp/Test/Semantic/Semantics/StaticNullChecking_TypeInference.cs

```diff
@@ -71,6 +74,192 @@ static void Main()
         }
 
         [Fact]
+        public void Default_TUnconstrained()
+        {
+            var source =
+@"class C
+{
+    static void F<T>()
+    {
+        var s = default(T);
+        s.ToString();
+        var t = default(T?);
+        t.ToString();
```

Comment:
Since no warning on `t.ToString()`, it looks like `default(T?)` is a `T!`. If that's the case, that seems surprising, language-wise.